### ISSUE TITLE
--runtime-tmpdir behavior strange - enhancement wish: extract only once to speed up --onefile

### ISSUE THREAD

**Issue by bufemc** [pyinstaller/pyinstaller#4994]
Update: I found out many stuff by myself, so you might skip A and directly jump to B.

A)

First of all, seems to be the best way to create an exe from a py, thanks a lot!

I created both, an onedir and an onefile variant. For a console application just doing something with sys.argv parameter..

I understand that the onefile variant needs to unpack first. In the documentation you say you could specify a --runtime-tmpdir .. however it seems not to work for me on Windows, or I am doing it wrong?

Plus, I miss this information: will it unpack every time, or only the first time? Many people complain about the slow startup time for the --onefile variant. So why not unpack only once, and for the second start just check if it was already unpacked and then skip this unpacking process. Is this already done, or not?

Another thing is that I have to exclude a lot of stuff, which I even do not use in my console application. Even in a very fresh venv without any modules installed there.

```
(venv) C:\workspace\python\handler>pyinstaller extractor.py --exclude asyncio --exclude socket --exclude bz2 --exclude hashlib --exclude lzma --exclude ssl --exclude select --exclude unicodedata --exclude pyexpat --onefile --runtime-tmpdir c:\workspace\python\handler\dist\tmp
```

As you see I pass it like this: ```--runtime-tmpdir c:\workspace\python\handler\dist\tmp```
I also tried to pass the path with normal slash /, but still, nothing in tmp folder after executing the one-file-exe.
I even created the tmp folder before, but did not help. No files.

B)

It would be quite bad if it's just unpacked, executed and deleted afterwards, then this option would not speed up things, which is wished by so many users, see e.g.:
https://github.com/pyinstaller/pyinstaller/issues/1782
https://github.com/pyinstaller/pyinstaller/issues/4980

This might be related to:
https://github.com/pyinstaller/pyinstaller/pull/4720

Update: I added ```print(f"The app is running in directory {sys._MEIPASS}")``` in the py file and get as output for 2 runs:

```
The app is running in directory c:\workspace\python\handler\dist\tmp\_MEI27162
The app is running in directory c:\workspace\python\handler\dist\tmp\_MEI74042
```

So I fear it's unpacked, but deleted after execution. So I might add the enhancement wish: why we can't extract in one specific folder, and just only once? It would speed up --onefile so much and make so many users happy ;-)

Later I added something to pause the app.. and now I see it's really unpacking the files there, but after execution deleting it.

Okay, I see the name tmpdir might include this meaning. So, what about a new option --runtime-permanent-unpackdir or something like that. BTW, I don't see how I could change the label, but you might change it to feature request.

**Comment by Legorooj** [comment]
This is incredibly complex, though I have a few ideas I will look into too see if it's feasable for me to implement.

**Comment by Legorooj** [comment]
Please see #4957, which we'll use for this.
